@grafana/ui: When to use Checkbox docs

jessover9000 opened the pull request:
This is a docs update. Added info about how checkboxes generally work and when to use another component instead.

jessover9000 committed:
Add "When to use" info to Checkbox.mdx

Added info about how checkboxes generally work and when to use another component instead.

jessover9000 committed:
Replaced words with proper component names

tskarhed commented on packages/grafana-ui/src/components/Forms/Checkbox.mdx:
```diff
@@ -5,6 +5,12 @@ import { Checkbox } from './Checkbox';
 
+### When to use
+
+`Checkbox` can be used in groups, where the single checkboxes have no dependencies. That means that selecting one doesn’t affect any other `Checkbox`. Checked represents true, un-checked represent false. So you can use them to select a binary option or multiple options in a set. When adding a description to your `Checkbox`, use positive commands, not negatives like "do not …".
```
`can` or `should` be used in group?

sarlinska replied:
1. I would start with:
"Checked represents true, un-checked represent false."
And then go about using them in groups.
2. Examples of positive commands would be useful – it's hard to imagine how they could sound. Show examples of do's and don'ts.

tskarhed commented on packages/grafana-ui/src/components/Forms/Checkbox.mdx:
```diff
@@ -5,6 +5,12 @@ import { Checkbox } from './Checkbox';
+
+`Checkbox` can be used in groups, where the single checkboxes have no dependencies. That means that selecting one doesn’t affect any other `Checkbox`. Checked represents true, un-checked represent false. So you can use them to select a binary option or multiple options in a set. When adding a description to your `Checkbox`, use positive commands, not negatives like "do not …".
+
+Checkboxes typically only trigger changes after sending a form. If your component should trigger a change immediately, it's better to use a toggle switch instead. Furthermore, checkboxes are not mutually exclusive. If you want to offer a mutually exclusive choice, use `RadioButtonGroup` or a `Select` dropdown.
```
Maybe use other wording than mutually exclusive or elaborate a bit on that.

sarlinska replied:
Again examples could be useful.

tskarhed reviewed (commented):
I get the feeling that checkboxes are used exclusively in submittable forms. Is that correct?

sarlinska reviewed (commented):
Just a couple of comments, mostly about adding some examples to better depict what we mean.

jessover9000 committed:
Update Checkbox.mdx

Elaborate on explanations. Add examples for how to use checkboxes the user-friendliest way.

tskarhed committed:
Merge branch 'master' into jessover9000-patch-1

dprokop commented on packages/grafana-ui/src/components/Forms/Checkbox.mdx:
```diff
@@ -5,6 +5,27 @@ import { Checkbox } from "./Checkbox";
+Checkboxes typically only trigger changes after sending a form. If your component should trigger a change immediately, it's better to use a toggle switch instead. Furthermore, checkboxes are not mutually exclusive. That means that selecting one will not disable the others or impact them in any other way. If you want to offer a mutually exclusive choice, use `RadioButtonGroup` or a `Select` dropdo ...
+
+**DO:**
+Show legends
+[ ] Blue
+[ ] Green
+[ ] Red
```
this example is weird to me:)

dprokop replied:
```suggestion
Show series
[ ] A-series
[ ] B-series
[ ] C-series
```

dprokop commented on packages/grafana-ui/src/components/Forms/Checkbox.mdx:
```diff
@@ -5,6 +5,27 @@ import { Checkbox } from "./Checkbox";
+[ ] Red
+
+**DON'T:**
+Show only
+[ ] Blue
+[ ] Green
+[ ] Red
```
```suggestion
Show only
[ ] A-series
[ ] B-series
[ ] C-series
```

tskarhed committed:
Apply suggestions from code review

Co-authored-by: Dominik Prokop <dominik.prokop@grafana.com>
